build: Build amd64 and arm64 multi-platform Docker image

chadlwilson opened the pull request:
## Description of Change

Uses docker buildx to build and push the docker image for `linux/arm64` as well as existing `linux/amd64`

- No changes to docker workflow secrets should be needed
- Implementation builds the image for both platforms, and then loads into docker for the native platform (`--load`)
- Adds testing of the docker image to the PR workflow so contributors can also tweak the docker build more safely
- Currently only runs docker tests for the "native" image for the platform, i.e `linux/amd64` on default runners.
  - If we want to specifically test the arm64 image, we could split to test on arm runners and/or force a test of the emulated images. However, it's a bit clunky with the current scripting based docker build (as opposed to using the docker actions that make this easier) - especially if you want to build once, and then test/push the exact same built image, rather than rebuilding.
- Since my local platform is arm64-native, I have run `test-docker.sh` locally on the arm64 image though

Sample images built and pushed using this approach: https://hub.docker.com/repository/docker/chadwilson/dependency-check/tags

See sample actions run at https://github.com/dependency-check/DependencyCheck/actions/runs/17936406550/job/51003624010?pr=7952

## Related issues

- fixes #6420 

## Have test cases been added to cover the new functionality?

yes

chadlwilson committed:
build: Build amd64 and arm64 multi-platform Docker image

Signed-off-by: Chad Wilson <29788154+chadlwilson@users.noreply.github.com>

jeremylong commented:
@chadlwilson can you add the test for the build failure that points to the setup instructions as you [mentioned above](https://github.com/dependency-check/DependencyCheck/pull/7952#issuecomment-3323891513)? I'm okay with the PR - I just think this would be helpful as we have people build things locally a lot and I want to avoid as many questions as we can.

chadlwilson commented:
Ack, will do shortly

chadlwilson committed:
build: Exit and print error if docker is not running with the containerd snapshotter

This is currently required to do a `docker buildx build --load` alongside multi-platform build, due to needing a convenient workaround for https://github.com/docker/buildx/issues/59#issuecomment-2612971318

Signed-off-by: Chad Wilson <29788154+chadlwilson@users.noreply.github.com>

chadlwilson committed:
build(deps): bump exec-maven-plugin from 1.3.1 to 3.5.1 within docker scripts

Signed-off-by: Chad Wilson <29788154+chadlwilson@users.noreply.github.com>

chadlwilson commented:
Rebased and added. Will now fail with:

```
Docker Engine is not running with the containerd snapshotter - this is currently needed to build and test ODC multi-platform images using docker buildx.
If using Docker Desktop, enable "Use containerd for pulling and storing images" per https://docs.docker.com/desktop/settings-and-maintenance/settings/#general
For more technical information on Docker Engine, see https://docs.docker.com/engine/storage/containerd/
```

marcelstoer reviewed (approved):
This is great, thanks!

jeremylong reviewed (approved):
LGTM
